pkg/status/kubefed.go

```diff
@@ -30,31 +28,23 @@ type KubefedAttributes struct {
 	Threshold      int64
 }
 
-// GetKubefedConditions uses the provided kubefed attributes to determine kubefed status
-func GetKubefedConditions(attrs KubefedAttributes) (*kubefed_v1beta1.KubeFedClusterStatus, error) {
+// GetKubefedConditions uses the provided kubefed attributes to determine status conditions
+func GetKubefedConditions(attrs KubefedAttributes) []toolchainv1alpha1.Condition {
 	// look up cluster connection status
 	fedCluster, ok := attrs.GetClusterFunc()
 	if !ok {
-		notFoundMsg := ErrMsgClusterConnectionNotFound
-		notFoundReason := toolchainv1alpha1.ToolchainStatusClusterConnectionNotFoundReason
-		badClusterStatus := kubefed_v1beta1.KubeFedClusterStatus{
-			Conditions: []kubefed_v1beta1.ClusterCondition{
-				{
-					Type:          kubefed_common.ClusterReady,
-					Status:        corev1.ConditionFalse,
-					Reason:        &notFoundReason,
-					Message:       &notFoundMsg,
-					LastProbeTime: metav1.Now(),
-				},
-			},
-		}
-		return &badClusterStatus, fmt.Errorf(notFoundMsg)
+		return []toolchainv1alpha1.Condition{*NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusClusterConnectionNotFoundReason, ErrMsgClusterConnectionNotFound)}
 	}
-	clusterStatus := *fedCluster.ClusterStatus.DeepCopy()
 
 	// check conditions of cluster connection
 	if !kubefed_util.IsClusterReady(fedCluster.ClusterStatus) {
-		return &clusterStatus, fmt.Errorf("the cluster connection is not ready")
+		for _, c := range fedCluster.ClusterStatus.Conditions {
+			if c.Type == "Ready" && c.Message != nil {
```

Comment:
Shouldn't we check the status of the Ready Condition instead of Message? If it's True or not.
We still need to make sure that message is not nil before using it. But we should not relay on the message as the status of the condition.
Check the status. And if it's not true then use the message if it's not nil. If it's nil then use our own message.

Reply:
Actually the `kubefed_util.IsClusterReady` check above already checks the status of the ready condition so this block will only execute in the case when Ready is false. So here I'm being extra cautious and checking if the kubefed message exists so we can reuse it or if we want to provide a generic message if one does not exist.

Reply:
OK. But then if status is false but there is no message, shouldn't we provide our own message instead?

Reply:
Yup, we do. Just outside this for loop we have
```
genericErrMsg := "the cluster connection is not ready"
		return []toolchainv1alpha1.Condition{*NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusClusterConnectionNotReadyReason, genericErrMsg)}
```
So we provide the message `"the cluster connection is not ready"`

Reply:
Right. I see it now.